src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState, useEffect } from 'react';
+import CircularProgress from '@mui/material/CircularProgress';
+import Tooltip from '@mui/material/Tooltip';
+import pick from 'lodash/pick';
+import groupBy from 'lodash/groupBy';
+import { useSnackbar } from 'notistack';
+import { format, fromUnixTime } from 'date-fns';
+import apiClient from 'api';
+import Location from 'types/location';
+import COUNTRY_CODES from 'components/LocationAutocomplete/countryCodes';
+import HOURS from './hours';
+
+interface ForecastBody {
+  dt: number;
+  main: {
+    humidity: number;
+    temp: number;
+  };
+  weather: [
+    {
+      description: string;
+      icon: string;
+    },
+  ];
+}
+
+interface Forecast {
+  list: ForecastBody[];
+}
+
+interface FormattedForecast
+  extends Array<[string, Pick<ForecastBody, 'dt' | 'main' | 'weather'>[]]> {}
+
+interface Props {
+  location: Location | null;
```

Comment:
Should be just `Location`. This component won't render if there is no selected location. We moved this check in `App`, recall?